### PR TITLE
(WIN-183) Add support for reading Stackify.json

### PR DESCRIPTION
**Pull request by marcdemz**

### CODE REVIEW

**Comment by homiedopie** [comment]
Added tests and updated some parts for the test to pass.

**Review by homiedopie — APPROVED** [review]

**Review by kbond-stackify — CHANGES_REQUESTED** [review]
This does not change the core configuration reading logic of StackifyLib in stackify-api-dotnet/Src/StackifyLib/Config.cs

**Comment by marcdemz** [comment]
> This does not change the core configuration reading logic of StackifyLib in stackify-api-dotnet/Src/StackifyLib/Config.cs

added support for ASPNETCORE_ENVIRONMENT and update reading logic

**Comment by marcdemz** [comment]
Updated to support DOTNET_ENVIRONMENT
updated support of reading settings from Stackify.json for LoadSettings.

**Comment by marcdemz** [comment]
added .net framework app
added support for IIS express

**Comment by homiedopie** [comment]
@marcdemz  - I think this looks good but we can put the reading of the file inside a separate function since it is being used for net core and full framework.

File: Src/StackifyLib/Config.cs [hunk 1]
```diff
@@ -197,5 +216,137 @@ internal static string Get(string key, string defaultValue = null)
...
+            catch (Exception ex)
+            {
+                StackifyAPILogger.Log("#Config #TryGetValue #Json failed", ex);
+            }
+
+            return r;
+        }
+        public static string GetEnvironment(JObject envName = null)
+        {
+            var ASPEnvironment = System.Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT");
```
Comment (kbond-stackify) [inline Src/StackifyLib/Config.cs:332]: These environment variables should be used to load additional overrides for the Stackify.json.  For example, if DOTNET_ENVIRONMENT is set to "Production" then both Stackify.json and Stackify.Production.json should be read and the values in Stackify.Production.json should replace those in Stackify.json (or add to to the values if the value was not in Stackify.json.  These are not intended to set the application environment to reporting to Retrace (at least not directly).  There are environment variables STACKIFY_APPNAME and STACKIFY_ENVIRONMENT that are used to override the AppName and Environment reported to Retrace supported in the wrappers, and these should also be supported in StackifyLib.

Comment (marcdemz) [reply Src/StackifyLib/Config.cs]: added support on which environment variable used to load the additional overrides for Stackify.json

**Pull request merged**